src/components/Faucet/Claim.tsx

```diff
@@ -1,34 +1,128 @@
-import { Button, Flex, Icon, useToast } from '@chakra-ui/react'
+import {
+  Button,
+  Flex,
+  Icon,
+  useToast,
+  Text,
+  Input,
+  FormControl,
+  FormHelperText,
+  FormErrorMessage,
+} from '@chakra-ui/react'
 import { ConnectButton } from '@rainbow-me/rainbowkit'
 import { errorToString, useClient } from '@vocdoni/react-providers'
 import { useEffect, useState } from 'react'
-import { useTranslation } from 'react-i18next'
-import { FaGithub } from 'react-icons/fa'
-import { useFaucet } from './use-faucet'
+import { Trans, useTranslation } from 'react-i18next'
+import { FaFacebook, FaGithub, FaGoogle } from 'react-icons/fa'
+import { signinUrlParams, useFaucet } from './use-faucet'
+import { ethers } from 'ethers'
+import { useForm } from 'react-hook-form'
+import { isValid } from 'date-fns'
+
+type ClaimProps = {
+  signinUrlParams?: signinUrlParams[]
+}
 
-export const Claim = () => {
+export const Claim = (props: ClaimProps) => {
   const { t } = useTranslation()
-  const { connected } = useClient()
+  const { account, connected, client } = useClient()
   const { loading, handleSignIn } = useClaim()
 
+  const [recipientAddress, setRecipientAddress] = useState<string>('')
+
+  const {
+    register,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      address: '',
+    },
+    mode: 'onChange',
+  })
+
+  useEffect(() => {
+    setRecipientAddress(account?.address as string)
+  }, [account])
```

Comment:
Is there any reason to specify the full `account` object? Better specify `account?.address`, since passing the entire object could cause undesired effect executions, thus causing unnecessary renders.